kani-driver/src/cbmc_output_parser.rs

```diff
@@ -811,16 +812,24 @@ fn has_check_failure(properties: &Vec<Property>, description: &str) -> bool {
 
 /// Replaces the description of all properties from functions with a missing
 /// definition.
-/// TODO: This hasn't been working as expected, see
-/// <https://github.com/model-checking/kani/issues/1424>
 fn modify_undefined_function_checks(mut properties: Vec<Property>) -> (Vec<Property>, bool) {
     let mut has_unknown_location_checks = false;
     for mut prop in &mut properties {
-        if prop.description.contains(ASSERTION_FALSE)
-            && extract_property_class(prop).unwrap() == DEFAULT_ASSERTION
+        if let Some(function) = &prop.source_location.function
+            && prop.description == DEFAULT_ASSERTION
             && prop.source_location.file.is_none()
         {
-            prop.description = "Function with missing definition is unreachable".to_string();
+            // Missing functions come with mangled names. `demangle` produces
+            // the demangled version if it's a mangled name, but for some reason
+            // the name is not being fully demangled. For example:
+            //
+            // function: _ZN60_$LT$alloc..string..String$u20$as$u20$core..clone..Clone$GT$5clone17h830b64d2ac32f613E
+            // demangle(function): <alloc::string::String as core::clone::Clone>::clone::h830b64d2ac32f613
+            //
+            // Note the last part, which doesn't appear to be demangled.
+            // Still, it's more readable than the mangled name.
+            let modified_description = format!("Function {} with missing definition is unreachable", demangle(function));
```

Comment:
Can you try `{:#}` instead of `{}`? I think the alternative formatting of `Demangle` should omit the hash at the end, which avoids the platform differences. I believe with that change, you can add the whole line to the `expected` files.

Reply:
`rustfilt` does something similar: https://github.com/luser/rustfilt/blob/3c81f107b73fdcf7bbb98b426c1214f15946ec90/src/main.rs#L47

Reply:
Thanks @fzaiser, that did the trick!